### DISCUSSION TITLE
Python CLI is not installed error message (in miniconda powershell with active env)

### DISCUSSION THREAD

**Discussion by delbert** [Azure/azure-dev#1837] [Q&A]
hello folks,

i am getting the following error message when i try to deploy a template:

```
ERROR: required external tools are missing:
 - Python CLI is not installed, please see https://wiki.python.org/moin/BeginnersGuide/Download to install
```

but i can get to the Python CLI just fine:

```
(env) PS C:\Users...> python
Python 3.10.9 | packaged by Anaconda, Inc. | (main, Mar  8 2023, 10:42:25) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

thanks for any hints,

delbert

**Comment by rajeshkamal5050** [comment]
@vhvb1989 @hemarina do we have any workaround for this? until we fix this #1623

**Reply by vhvb1989** [reply]
Installing Python for windows from: https://www.python.org/downloads/ would be the workaround. That's the suggested action from azd's error as well.

Another workaround is to create an alias from `py` to `python`. For example, if you are using PowerShell: `Set-Alias py python` would solve this issue on the current console.

**Reply by delbert** [reply]
thanks for this suggestion, but i still run into the same error.  this what i tried:

```
(oai) PS C:\Users\...> Set-Alias py python
(oai) PS C:\Users\...> azd up

Initializing a new project (azd init)

Provisioning Azure resources (azd provision)
Provisioning Azure resources can take some time

ERROR: required external tools are missing:
 - Python CLI is not installed, please see https://wiki.python.org/moin/BeginnersGuide/Download to install
```

but, if i try the command py, i end up in the interpreter, just as we would expect:

```
(oai) PS C:\Users\...> py
Python 3.10.9 | packaged by Anaconda, Inc. | (main, Mar  8 2023, 10:42:25) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()
(oai) PS C:\Users\...> 
```

am i missing something?  thanks

**Reply by vhvb1989** [reply]
It looks like `azd` is expecting to find `py` within System path :(
That's why setting the `alias` is not enough here. My apologies for providing incorrect workaround.

Please try installing Python for windows from: https://www.python.org/downloads/windows/
It should install `py` inside %SystemRoot%

**Comment by delbert** [comment]
thanks, Victor.

a little under three weeks ago, i did precisely what you recommend and torqued my system so badly, it has taken me this long to recover.  i rely heavily on Conda virtual environments for my work.

delbert
________________________________

**Reply by lukas-pfeifhofer** [reply]
I had the same issue, in my case the problem was that python was installed from Microsoft Store like an App, and in this case python is not recognized by the system PATH. After uninstalling the App, and reinstalling python by the MSI file downloaded from [python.org](https://www.python.org) , the deployment was able to recognize python.exe through system PATH, and everything works fine for me.

Lukas
